blockchain/Validation.go

```diff
@@ -1594,66 +1594,63 @@ func ConnectBlock(param *msg.BitcoinParams, pblock *model.Block, state *model.Va
 	//sc.Lock()
 	//defer sc.Unlock()
 
-	nTimeStart := utils.GetMockTimeInMicros()
+	nTimeStart := utils.GetMicrosTime()
 
 	// Check it again in case a previous version let a bad block in
 	if !CheckBlock(param, pblock, state, !fJustCheck, !fJustCheck) {
-		log.Error("CheckBlock: %s", FormatStateMessage(state))
+		return logger.ErrorLog(fmt.Sprintf("CheckBlock: %s", FormatStateMessage(state)))
 	}
 
 	// Verify that the view's current state corresponds to the previous block
-	var hashPrevBlock utils.Hash
-	if pindex.PPrev == nil {
-		hashPrevBlock = utils.Hash{}
-	} else {
-		hashPrevBlock = pindex.PPrev.GetBlockHash()
-	}
+	hashPrevBlock := pindex.PPrev.GetBlockHash()
 
 	if hashPrevBlock != view.GetBestBlock() {
 		panic("error: hashPrevBlock not equal view.GetBestBlock()")
 	}
 
 	// Special case for the genesis block, skipping connection of its
 	// transactions (its coinbase is unspendable)
-	if &pblock.Hash == param.GenesisHash {
+	if pblock.Hash.IsEqual(param.GenesisHash) {
 		if !fJustCheck {
 			view.SetBestBlock(pindex.GetBlockHash())
 		}
 		return true
 	}
 
 	fScriptChecks := true
-	if &HashAssumeValid != nil {
+	if HashAssumeValid != utils.HashZero {
 		// We've been configured with the hash of a block which has been
 		// externally verified to have a valid history. A suitable default value
 		// is included with the software and updated from time to time. Because
 		// validity relative to a piece of software is an objective fact these
 		// defaults can be easily reviewed. This setting doesn't force the
 		// selection of any particular chain but makes validating some faster by
 		// effectively caching the result of part of the verification.
-		it := MapBlockIndex.Data[HashAssumeValid]
-		if it.GetAncestor(pindex.Height) == pindex && gpindexBestHeader.GetAncestor(pindex.Height) == pindex &&
-			gpindexBestHeader.ChainWork.Cmp(&param.MinimumChainWork) > 0 {
-			// This block is a member of the assumed verified chain and an
-			// ancestor of the best header. The equivalent time check
-			// discourages hashpower from extorting the network via DOS
-			// attack into accepting an invalid block through telling users
-			// they must manually set assumevalid. Requiring a software
-			// change or burying the invalid block, regardless of the
-			// setting, makes it hard to hide the implication of the demand.
-			// This also avoids having release candidates that are hardly
-			// doing any signature verification at all in testing without
-			// having to artificially set the default assumed verified block
-			// further back. The test against nMinimumChainWork prevents the
-			// skipping when denied access to any chain at least as good as
-			// the expected chain.
-			fScriptChecks = (GetBlockProofEquivalentTime(gpindexBestHeader, pindex, gpindexBestHeader, param)) <= 60*60*24*7*2
-		}
-	}
-
-	nTime1 := utils.GetMockTimeInMicros()
+		if MapBlockIndex.Data[HashAssumeValid] != nil {
```

Comment:
why do you not use if v , ok := MapBlockIndex.Data[HashAssumeValid]; ok{
.....
}

Reply:
fix alerady